docs/DeveloperGuide.md

```diff
@@ -219,18 +219,141 @@ storage.saveAllData();
+* *settleUp*: Settles the debt of the specified member by creating a transaction with the lender(s) based on the transaction solution of the group.
+* *saveAllData*: Saves the current member and transaction data of the group to the storage handler.
+* *listDebts*: Returns a string representation of the current transaction solution to all debts in the group.
+* *listIndivDebt*: Returns a string representation of the current transaction solution to the debt of a specified member in the group.
+
+The `GroupList` class has the following key methods.
+* *switchActiveGroup*: Switches the active group to the group with the specified name. This method is used when the user wants to switch to manage a different group.
+* *createGroup*: Prompts user to enter a new group name and creates a new group with the specified name. Automatically sets it as the active group.
+* *loadGroupList*: Loads the list of groups stored in the application from the storage handler.
+* *addGroup*: Adds a group to the group list. This method is used when a new group is created.
+* *deleteGroup*: Deletes a group from the group list based on the specified group name. The member and transaction files associated with the group are also deleted from storage.
+* *saveGroupList*: Saves the list of groups stored in the groupList to the storage handler.
+
 <ins>Usage Example</ins>
 
+The following code segment outlines a sample use of `Group`.
+
+```
+import longah.util.TransactionList;
+import longah.handler.UI;
+import longah.exception.LongAhException;
+import longah.node.Group;
+
+// Creating a new group
+Group myGroup = new Group("MyGroup");
+
+// Adding members to the group
+myGroup.getMemberList().addMember("Alice");
+myGroup.getMemberList().addMember("Bob");
+myGroup.getMemberList().addMember("Charlie");
+
+// Adding transactions to the group
+TransactionList transactions = new TransactionList();
+transactions.addTransaction("Alice p/Bob a/20", myGroup.getMemberList());
+transactions.addTransaction("Charlie p/Alice a/15", myGroup.getMemberList());
+transactions.addTransaction("Bob p/Charlie a/10", myGroup.getMemberList());
+
+// Setting the transactions for the group
+myGroup.setTransactionList(transactions);
+
+// Updating transaction solutions for the group
+myGroup.updateTransactionSolution();
+
+// Listing debts in the group
+String debts = myGroup.listDebts();
+UI.showMessage("Debts in the group:\n" + debts);
+
+// Settling up debts for a specific member
+myGroup.settleUp("Alice");
+
+// Saving all data related to the group
+myGroup.saveAllData();
+```
+
+The following code segment outlines a sample use of `GroupList`.
+
+```
+import longah.exception.LongAhException;
+import longah.handler.UI;
+import longah.node.GroupList;
+
+// Creating a new GroupList instance
+GroupList groupList = new GroupList();
+
+// Creating a new group and adding it to the group list
+groupList.createGroup();
+
+// Adding more groups to the list
+groupList.createGroup();
```

Comment:
Now that I look at this it looks rather... interesting. Would suggest either removing these 3 lines or adding a method to `GroupList` with String as its param and using that here

Reply:
Considering `createGroup()` doesnt even execute if the grouplist is empty this just seems erroneous actually

Reply:
changing to `addGroup` in next commit